Gems/PhysX/Code/Tests/Benchmarks/PhysXBenchmarksUtilities.cpp

```diff
@@ -65,21 +81,69 @@ namespace PhysX::Benchmarks
+                if (benchmarkObjectType == PhysX::Benchmarks::RigidBodyApiObject)
+                {
+                    AzPhysics::Scene* scene = AZ::Interface<AzPhysics::SystemInterface>::Get()->GetScene(sceneHandle);
+                    AzPhysics::SimulatedBodyHandle simBodyHandle = scene->AddSimulatedBody(&rigidBodyConfig);
+
+                    AZStd::get_if<AzPhysics::SimulatedBodyHandleList>(&benchmarkRigidBodies)->push_back(simBodyHandle);
+                }
+                else if (benchmarkObjectType == PhysX::Benchmarks::RigidBodyEntity)
+                {
+                    EntityPtr entity;
+                    if (rigidBodyConfig.m_entityId.IsValid())
+                    {
+                        entity = AZStd::make_shared<AZ::Entity>(rigidBodyConfig.m_entityId, "TestEntity");
+                    }
+                    else
+                    {
+                        entity = AZStd::make_shared<AZ::Entity>("TestEntity");
+                    }
+
+                    auto* transformComponent = entity->CreateComponent<AzFramework::TransformComponent>();
+                    transformComponent->SetWorldTM(
+                        AZ::Transform::CreateFromQuaternionAndTranslation(rigidBodyConfig.m_orientation, rigidBodyConfig.m_position));
+
+                    auto boxColliderComponent = entity->CreateComponent<PhysX::BoxColliderComponent>();
+                    boxColliderComponent->SetShapeConfigurationList({ AzPhysics::ShapeColliderPair(rigidBodyColliderConfig, shapeConfig) });
+
+                    entity->CreateComponent<PhysX::RigidBodyComponent>(rigidBodyConfig, sceneHandle);
+
+                    entity->Init();
+                    entity->Activate();
+
+                    AZStd::get_if<PhysX::EntityList>(&benchmarkRigidBodies)->push_back(entity);
+                }
             }
 
-            return rigidBodies;
+            return benchmarkRigidBodies;
         }
 
-        AZStd::vector<AzPhysics::RigidBody*> GetRigidBodiesFromHandles(AzPhysics::Scene* scene, const AzPhysics::SimulatedBodyHandleList& handlesList)
+        AZStd::vector<AzPhysics::RigidBody*> GetRigidBodiesFromHandles(
+            AzPhysics::Scene* scene, const Utils::BenchmarkRigidBodies& benchmarkRigidBodies)
         {
             AZStd::vector<AzPhysics::RigidBody*> rigidBodies;
-            rigidBodies.reserve(handlesList.size());
-            for (auto handle : handlesList)
+
+            if (auto handlesList = AZStd::get_if<AzPhysics::SimulatedBodyHandleList>(&benchmarkRigidBodies))
             {
-                rigidBodies.push_back(azdynamic_cast<AzPhysics::RigidBody*>(scene->GetSimulatedBodyFromHandle(handle)));
+                rigidBodies.reserve(handlesList->size());
+
+                for (const AzPhysics::SimulatedBodyHandle& handle : *handlesList)
+                {
+                    rigidBodies.push_back(azdynamic_cast<AzPhysics::RigidBody*>(scene->GetSimulatedBodyFromHandle(handle)));
+                }
```

Comment:
This can also use `AZStd::views::transform` and AZStd::from_range constructor for AZStd::vector to copy over the rigid body pointers

```suggestion
                rigidBodies = AZStd::vector<AzPhysics::RigidBody*>(AZStd::from_range, *handlesLists | AZStd::views::transform([&scene](const AzPhysics::SimulatedBodyHandle& handle) { azrtti_cast<AzPhysics::RigidBody*>(scene->GetSimulatedBodyFromHandle(handle)); });

                for (const AzPhysics::SimulatedBodyHandle& handle : *handlesList)
                {
                    rigidBodies.push_back(azdynamic_cast<AzPhysics::RigidBody*>(scene->GetSimulatedBodyFromHandle(handle)));
                }
```